scan.c

```diff
@@ -129,8 +131,25 @@ struct iio_scan_context * iio_create_scan_context(
 	if (!backend || strstr(backend, "local"))
 		ctx->scan_local = true;
 
-	if (!backend || strstr(backend, "usb"))
+	if (!backend || strstr(backend, "usb")) {
 		ctx->scan_usb = true;
+		if (backend) {
+			char *ptr;
+			ptr = strstr(backend, "vid:");
```

Comment:
I don't like strstr - they check the whole string and are generally not a good idea when parsing strings.
If you strip the URI header (`usb:`) and get the parameters in a string pointer `args`, you could do:
```
if (args) {
    ret = iio_sscanf(args, "vid=04x;pid=04x", &ctx->scan_usb_vid, &ctx->scan_usb_pid);
    /* ret == 1: only VID, ret == 2: VID+PID, other values (including 0): parsing error */
}
```

Reply:
there are still a few strstr around, but it's no more than were there before...